src/wpool/mongoose_wpool_mgr.erl

```diff
@@ -70,7 +70,13 @@ start_link(Type) ->
 
 start(Type, Host, Tag, PoolOpts, ConnOpts) ->
     ok = ensure_started(Type),
-    gen_server:call(name(Type), {start_pool, Host, Tag, PoolOpts, ConnOpts}).
+    try
+        gen_server:call(name(Type), {start_pool, Host, Tag, PoolOpts, ConnOpts})
+    catch
+        _:_ ->
+            %% hide connection options
```

Comment:
I'm afraid we are hiding to much here. How do we know what was the reason and how to debug the issue if we only say that the pool failed to start?

Reply:
first of all, in the most cases `gen_server:call()` fails with one of 2 reasons:
- timeout 
- gen_server dies before responding

in both cases you will rather want to debug the corresponding gen_server. not the caller.

but if you really need the exact error you can still get it using dbg w/o changing the code, see the example:
```
Eshell V10.6  (abort with ^G)
1> dbg:tracer().
{ok,<0.80.0>}
2> dbg:p(all,call).
{ok,[{matched,nonode@nohost,40}]}
3> dbg:tp(gen_server, call,[{[some_server,'_'],[],[{exception_trace}]}]).
{ok,[{matched,nonode@nohost,2},{saved,1}]}
4> catch gen_server:call(some_server,test).
(<0.78.0>) call gen_server:call(some_server,test)
(<0.78.0>) exception_from {gen_server,call,2} {exit,{noproc,{gen_server,call,[some_server,test]}}}
{'EXIT',{noproc,{gen_server,call,[some_server,test]}}}
5> catch gen_server:call(another_server,test).
{'EXIT',{noproc,{gen_server,call,[another_server,test]}}}
6>
```

Reply:
unfortunately gen_server doesn't declare the format of the error, so there's no warranty that it stays unchanged from one erlang release to another

Reply:
@DenysGonchar 

> but if you really need the exact error you can still get it using dbg w/o changing the code, see the example

That's the point. In many situations we don't have access to the node, so we are not able to run `dbg` yet we are expected to help people.

Reply:
change rolled back, the further work in this area to be scheduled for the next release(s).